server/platform/shared/filestore/s3store.go

```diff
@@ -368,6 +368,60 @@ func (b *S3FileBackend) CopyFile(oldPath, newPath string) error {
 	return nil
 }
 
+// DecodeFilePathIfNeeded is a special method to URL decode all older
+// file paths. It is only needed for the migration, and will be removed
+// as soon as the migration is complete.
+func (b *S3FileBackend) DecodeFilePathIfNeeded(path string) error {
+	// Encode and check if file path changes.
+	// If there is no change, then there is no need to do anything.
+	if path == s3utils.EncodePath(path) {
+		return nil
+	}
+
+	// Check if encoded path exists.
+	exists, err := b.lookupOriginalPath(s3utils.EncodePath(path))
```

Comment:
nit: original here means "the new path", right? Should we clarify this with some variable names, e.g. `doublyEscapedPath` and `correctPath`?